apis/src/main/java/xyz/bookself/controllers/user/BookListController.java

```diff
@@ -57,41 +57,35 @@ public BookListController(BookselfApiConfiguration configuration, BookListReposi
         return new ResponseEntity<>(booksInList, HttpStatus.OK);
     }
 
-    /**
-     * Update a book list (shelf).
-     *  - Rename shelf
-     *  - Add books
-     *  - Remove books
-     * Check if fields are not null before updating shelf.
-     * Unrecognized fields are ignored.
-     * @param shelfDto JSON of the format:
-     *      {
-     *          "newListName": "New Name",
-     *          "booksToBeAdded": [ "book-id-1", "book-id-2" ],
-     *          "booksToBeRemoved": [ "book-id-3", "book-id-4" ]
-     *      }
-     * @param id the shelf id
-     * The correct request method for updating is PUT
-     */
-    @PutMapping(value = "/{id}/update", consumes = MediaType.APPLICATION_JSON_VALUE)
-    public ResponseEntity<BookList> renameShelf(@PathVariable String id, @RequestBody ShelfDto shelfDto) {
-
-        final String newListName = shelfDto.getNewListName();
-        final Set<String> booksToBeAdded = shelfDto.getBooksToBeAdded();
-        final Set<String> booksToBeRemoved = shelfDto.getBooksToBeRemoved();
-        final String newBookListId = shelfDto.getNewBookListId();
+    @PutMapping(value = "/{id}/update-list-name", consumes = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<BookList> renameShelf(@PathVariable String id, @RequestBody ShelfNameDTO shelfNameDTO) {
```

Comment:
If you decide to use `@NotBlank` in the DTO like suggested below, make sure to add `@Valid` before `@RequestBody` so it's enforced.